tests/pos/i756.scala

```diff
@@ -0,0 +1,10 @@
+class Test[T] {
+
+  def f(x: Int) = try {
+    ???
+  }
+  catch {
+    case ex: scala.runtime.NonLocalReturnControl[T @scala.unchecked] =>
```

Comment:
This should probably be a BytecodeTest that checks that we don't use the instanceof opcode: https://github.com/lampepfl/dotty/blob/master/compiler/test/dotty/tools/backend/jvm/DottyBytecodeTests.scala